proxy/reverse.go

```diff
@@ -118,6 +121,9 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 		redirectRequest(proxyreq, ep.URL)
 
 		res, err = p.transport.RoundTrip(proxyreq)
+		if requestCanceled {
```

Comment:
this seem like a race. if the client closed the connection and the proxy's tr finished the request at the same time, two routines can access requestCanceled at the same time.

Will a canceled request return a special error?